src/Metadata/Factory.php

```diff
@@ -50,7 +50,7 @@ public function load(string $role): ?MetadataBase
                     $currentMetadata = TimestampMetadata::createFromJson($json);
                     break;
                 default:
-                    $currentMetadata = TargetsMetadata::createFromJson($json);
+                    $currentMetadata = TargetsMetadata::createFromJson($json, $role);
```

Comment:
This is the only unrelated fix. We probably need test coverage for the `Factory` class in another PR. 

the 'targets' type is the only  type where role doesn't always equal type. 

without this fix `$delegatedTargetsMetadata->getRole()` would always equal 'targets' where it should equal the delegated role name.